### DISCUSSION TITLE
Video stream url

### DISCUSSION THREAD

**Discussion by wociscz** [Lillifee/raspiCam#40] [General]
Hi, is it possible to get url of h264 stream out of raspicam? 
I mean to process the stream somewhere else, like run it in VLC, or process the stream in frigate, stream on youtube or so...

**Comment by Lillifee** [comment]
Hey @wociscz 

sure http://your_raspi_ip:8000
- **/api/stream/mjpeg** mjpeg stream
- **/api/stream/live**  h264 stream

**Reply by wociscz** [reply]
Sweet, thanks!

**Reply by psylover78** [reply]
Hi, this down't seem to work for me with this url http://192.168.68.108:8000/api/stream... Any idea what I/m doing wrong? really LOVE the app!!!!!

**Reply by Lillifee** [reply]
Hey @psylover78 

What did you tried so far? 
There is only one type of stream (h264, mjpeg) active at the same time. For the h264 stream, you need a player like VLC. The mjpeg stream should work in the browser, but you have to change it in the stream settings. I can try it with the latest version tomorrow.

**Reply by Lillifee** [reply]
Hey @psylover78 

i tried it with the latest version and the H264 doesn't work in VLC. 
The mjpeg works as expected if you change the settings to mjpeg:

![image](https://user-images.githubusercontent.com/23337626/223176209-ce8cd159-9eda-4467-ba75-6f963eaf53f1.png)

and then open the http://your_raspi_ip:8000/api/stream/mjpeg

In one of the last releases, I tried to improve the streaming to multiple clients.
I will try to invest further into the problem and let you know.

**Reply by james-dmu** [reply]
I didn't seem to be able to get h264 working either.

However I was going to ask something similar as to if the preview could possibly stay active when recording video as it's handy to be able to see what is being recorded. 

Currently I am using raspiCam to record some bats using my IR camera however I have no idea what I have captured until I review the footage later.

**Reply by Lillifee** [reply]
Hey @james-dmu,

this is a different topic and i add a new feature request.
https://github.com/Lillifee/raspiCam/issues/57

I already thought about this since the beginning. Currently, libcamera-vid only supports a single stream at a time. This stream is recorded to a file during recordings and streamed live with different settings on the website. However, if we attempt to preview a 1080p video with a high bitrate, it will cause delays and lags in the live stream. Consequently, it becomes quite challenging to simultaneously record a high-quality video and stream it. In my opinion, the most viable solution would be to save the high-quality video to a file, compress it using ffmpeg, and then stream the compressed version on the website. However, implementing this solution may require significant refactoring since raspicam does not currently utilize ffmpeg.
